croncat/src/rpc/signer.rs

```diff
@@ -17,150 +19,134 @@ use cw_rules_core::msg::QueryConstruct;
+        cfg: ChainConfig,
         manager: String,
         key: bip32::XPrv,
-        gas_prices: f32,
-        gas_adjustment: f32,
+        mnemonic: String,
     ) -> Result<Self, Report> {
         // TODO: How should we handle this? Is the hack okay?
         // Quick hack to add https:// to the url if it is missing
-        let grpc_url = if !grpc_url.starts_with("https://") {
-            format!("https://{}", grpc_url)
-        } else {
-            grpc_url
-        };
         let rpc_url = if !rpc_url.starts_with("https://") {
             format!("https://{}", rpc_url)
         } else {
             rpc_url
         };
 
-        let client = timeout(
-            Duration::from_secs(10),
-            CosmosFullClient::new(
-                rpc_url,
-                grpc_url,
-                chain_info,
-                key,
-                gas_prices,
-                gas_adjustment,
-            ),
-        )
-        .await??;
-        let account_id = client
-            .key()
+        // Get the account id from the key.
+        let signing_key: SigningKey = key.into();
+        let account_id = signing_key
             .public_key()
-            .account_id(&client.chain_info.bech32_prefix)?;
+            .account_id(&cfg.info.bech32_prefix)?;
+
+        // Create a new RPC client
+        let mut rpc_client = RpcClient::new(&cfg, rpc_url.as_str())?;
+        // TODO: This is a hack to get around the fact that cosm-tome doesn't
+        // let us pass an xprv.
+        rpc_client.set_mnemonic(mnemonic);
+        rpc_client.set_denom(&cfg.info.fees.fee_tokens[0].denom);
 
         Ok(Self {
-            client,
             account_id,
             manager,
+            rpc_client,
         })
     }
 
     pub fn from_chain_config(
         chain_config: &ChainConfig,
         key: bip32::XPrv,
+        mnemonic: String,
     ) -> impl Future<Output = Result<Self, Report>> {
-        GrpcSigner::new(
+        Signer::new(
             chain_config.info.apis.rpc[0].address.clone(),
-            chain_config.info.apis.grpc[0].address.clone(),
-            chain_config.info.clone(),
+            chain_config.clone(),
             chain_config.manager.clone(),
             key,
-            chain_config.gas_prices,
-            chain_config.gas_adjustment,
+            mnemonic,
         )
     }
 
-    pub async fn query_croncat<T>(&self, msg: &QueryMsg) -> Result<T, Report>
+    pub async fn query_croncat<R, S>(&self, msg: S) -> Result<R, Report>
     where
-        T: DeserializeOwned,
+        S: Serialize,
+        R: DeserializeOwned,
     {
-        let out = timeout(
-            Duration::from_secs(30),
-            self.client
-                .query_client
-                .query_contract(&self.manager.to_string(), msg),
-        )
-        .await
-        .map_err(|err| eyre!("Timeout (30s) while querying contract: {}", err))??;
+        let out = timeout(Duration::from_secs(30), self.rpc_client.wasm_query(msg))
```

Comment:
This likely needs a shorter configurable timeout. Chain blocks are 6 seconds, so 30seconds is too far behind